logistics/scripts/modules/settings/settings-management.psm1

```diff
@@ -0,0 +1,351 @@
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Admin'])    = "Server=(local); Trusted_Connection=True; Database=EdFi_Admin;"
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Security']) = "Server=(local); Trusted_Connection=True; Database=EdFi_Security; Persist Security Info=True;"
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Master'])   = "Server=(local); Trusted_Connection=True; Database=master;"
+            }
+            DatabaseTemplate  = @{
+                MinimalTemplateScript   = 'EdFiMinimalTemplate'
+                PopulatedTemplateScript = 'GrandBend'
+            }
+        }
+        PostgreSQL = @{
+            ConnectionStrings = @{
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Ods'])      = "Host=localhost; Port=5432; Username=postgres; Database=EdFi_{0};"
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Admin'])    = "Host=localhost; Port=5432; Username=postgres; Database=EdFi_Admin;"
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Security']) = "Host=localhost; Port=5432; Username=postgres; Database=EdFi_Security;"
+            }
+            DatabaseTemplate  = @{
+                MinimalTemplateScript   = 'PostgreSQLMinimalTemplate'
+                PopulatedTemplateScript = 'PostgreSQLPopulatedTemplate'
+            }
+        }
+    }
+}
+
+function Get-SubtypesByFeature {
+    return @{
+        changeQueries               = 'changes'
+        ownershipBasedAuthorization = 'RecordOwnership'
+    }
+}
+
+function Test-AppSettings {
+    param(
+        [Parameter(ValueFromPipeline = $true)]
+        [string] $Path
+    )
+
+    if (-not (Test-Path $Path)) { return $false }
+
+    try {
+        Get-Content $Path | ConvertFrom-Json | Out-Null
+        return $true
+    }
+    catch {
+        return $false
+    }
+}
+
+function Assert-ValidAppSettings([string[]] $SettingsFiles = (Get-ChildItem "$(Get-RepositoryRoot 'Ed-Fi-ODS-Implementation')\**\appsettings*.json" -Recurse)) {
+    $result = @()
+
+    foreach ($file in $SettingsFiles) {
+        if (Test-AppSettings $file) {
+            $result += @{ file = $file; success = $true }
+        }
+        else {
+            try {
+                Get-Content $file | ConvertFrom-Json
+            }
+            catch {
+                $result += @{ file = $file; success = $false; exception = $_ }
+            }
+        }
+    }
+
+    return $result
+}
+
+function Add-ApplicationNameToConnectionStrings([hashtable] $ConnectionStrings = @{ }, [string] $ApplicationName) {
+    $newConnectionStrings = Merge-Hashtables @{ }, $ConnectionStrings
+
+    foreach ($key in $ConnectionStrings.ConnectionStrings.Keys) {
+        if (-not ($ConnectionStrings.ConnectionStrings[$key] -like '*Application Name*')) {
+            $newConnectionStrings.ConnectionStrings[$key] += " Application Name=$ApplicationName;"
+        }
+    }
+
+    return $newConnectionStrings
+}
+
+function Get-ProjectPath {
```

Comment:
This information seems like it should be baked into the object you're returning with Get-DevelopmentSettingsByProject, rather than using convention to try to find them.

Reply:
removed and added paths to `Get-DefaultDevelopmentSettingsByProject`